### PR TITLE
Port async support from beta to the stable channel

### PR DESCRIPTION
**Pull request by richardm-stripe**
Created via `git checkout -p beta <files that weren't codegenned>` and some judicious manual cleaning up. I cherry-picked the PRs from outside collaborators for attribution reasons.

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -1,6 +1,14 @@
 # These requirements must be installable on all our supported versions
+
+# This is the last version of httpx compatible with Python 3.6
+httpx == 0.22.0
```
Comment (pakrym-stripe) [inline test-requirements.txt:4]: should we test on later version when python_version > "3.6"?

File: stripe/_api_requestor.py [hunk 2]
```diff
@@ -538,6 +617,90 @@ def request_raw(
 
         return rcontent, rcode, rheaders
 
+    async def request_raw_async(
```
Comment (pakrym-stripe) [inline stripe/_api_requestor.py:620]: nit: remove?

File: stripe/_api_resource.py [hunk 3]
```diff
@@ -90,12 +117,10 @@ def _request(
...
-    # The `method_` and `url_` arguments are suffixed with an underscore to
-    # avoid conflicting with actual request parameters in `params`.
     def _request_and_refresh(
         self,
-        method_: Literal["get", "post", "delete"],
-        url_: str,
+        method: Literal["get", "post", "delete"],
+        url: str,
         params: Optional[Mapping[str, Any]] = None,
         _usage: Optional[List[str]] = None,
```
Comment (pakrym-stripe) [inline stripe/_api_resource.py:125]: nit: if you are renaming parameters might as well rename the `_usage`, it's on a private method now.

**Review by pakrym-stripe — APPROVED** [review]
A few comments.

File: stripe/_api_requestor.py (hunk 2 above)
Comment (richardm-stripe) [reply stripe/_api_requestor.py:620]: No, `request_raw` is an internal helper function for `request`, it is not `raw_request`, and we need the asynchronous version.

Comment (richardm-stripe) [reply stripe/_api_requestor.py:620]: We should consider renaming

File: stripe/_api_resource.py (hunk 3 above)
Comment (richardm-stripe) [reply stripe/_api_resource.py]: Done in [rename _usage -> usage](https://github.com/stripe/stripe-python/pull/1288/commits/1436ce40384b1a9ce7e5a936cf2133131a31004b)

**Comment by richardm-stripe** [comment]
cc @pakrym-stripe want to continue to talk through StripeStreamResponseAsync to see if my thoughts are reasonable

File: stripe/__init__.py [hunk 4]
```diff
@@ -134,6 +134,7 @@ def set_app_info(
 from stripe._stripe_response import StripeResponseBase as StripeResponseBase
 from stripe._stripe_response import (
     StripeStreamResponse as StripeStreamResponse,
+    StripeStreamResponseAsync as StripeStreamResponseAsync,
```
[the remark this answers is not in the document]
Comment (pakrym-stripe) [reply stripe/__init__.py:137]: Yeah, I think your reasoning makes sense, there is not way to implement both sets of methods on the same class.

**Review by pakrym-stripe — APPROVED** [review]

**Pull request merged**